hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieTestTable.java

```diff
@@ -177,10 +161,14 @@ public HoodieTestTable addDeltaCommit(String instantTime) throws Exception {
     return this;
   }
 
-  public HoodieTestTable addReplaceCommit(String instantTime, HoodieRequestedReplaceMetadata requestedReplaceMetadata, HoodieReplaceCommitMetadata metadata) throws Exception {
+  public HoodieTestTable addReplaceCommit(
+      String instantTime,
+      HoodieRequestedReplaceMetadata requestedReplaceMetadata,
+      HoodieReplaceCommitMetadata completeReplaceMetadata,
+      HoodieCommitMetadata inflightReplaceMetadata) throws Exception {
```

Comment:
can we set HoodieCommitMetadata to Option< HoodieCommitMetadata >  to avoid null

Reply:
hmm, if you track it down to `createInflightReplaceCommit` where `HoodieCommitMetadata` is being referenced, a `null` check is being presented there. However, I do agree an `Optional` reminds people of a null check though the
`o.isPresent()` check is _hardly_ any better than `o != null`. I could make the change and also update `createRequestedReplaceCommit` to adopt the same Optional `HoodieRequestedReplaceMetadata`.